packages/service-clients/end-to-end-tests/azure-client/src/test/containerCreate.spec.ts

```diff
@@ -516,5 +531,76 @@ for (const testOpts of testMatrix) {
 			const result = containerCurrent1.initialObjects.map1.get<string>("key");
 			assert.strictEqual(result, "value", "Value not found in copied container");
 		});
+
+		for (const compatibilityMode of ["1", "2"] as const) {
+			it(`op grouping works as expected (compatibilityMode: ${compatibilityMode})`, async function () {
+				const { container: container1 } = await clientCurrent1.createContainer(
+					schemaCurrent,
+					compatibilityMode,
+				);
+				const containerId = await container1.attach();
+
+				if (container1.connectionState !== ConnectionState.Connected) {
+					await timeoutPromise((resolve) => container1.once("connected", () => resolve()), {
+						durationMs: connectTimeoutMs,
+						errorMsg: "container connect() timeout",
+					});
+				}
+
+				const containerProcessSpy = sandbox.spy(
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+					(container1 as any).container,
+					"processRemoteMessage",
+				);
+
+				// Explicitly force ops sent to be in the same batch
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+				(container1 as any).container._runtime.orderSequentially(() => {
+					const map1 = container1.initialObjects.map1;
+					map1.set("1", 1);
+					map1.set("2", 2);
+					map1.set("3", 3);
+				});
+
+				const { container: container2 } = await clientCurrent1.getContainer(
+					containerId,
+					schemaCurrent,
+					compatibilityMode,
+				);
+				const map2 = container2.initialObjects.map1;
+
+				// Process ops coming from service
+				assert.strictEqual(await mapWait(map2, "1"), 1);
+				assert.strictEqual(await mapWait(map2, "2"), 2);
+				assert.strictEqual(await mapWait(map2, "3"), 3);
+
+				// Inspect the incoming ops
+				let groupedBatchCount = 0;
+				for (const call of containerProcessSpy.getCalls()) {
+					const message = call.firstArg as ISequencedDocumentMessage;
+					if (
+						message.type === MessageType.Operation &&
+						// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+						(message.contents as any).type === "groupedBatch"
```

Comment:
You can avoid the `any` and eslint complaints like this if you want:

```suggestion
						(message.contents as {type: string}).type === "groupedBatch"
```